File: 1379 - Find a Corresponding Node of a Binary Tree in a Clone of That Tree/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    TreeNode* getTargetCopy(TreeNode* A, TreeNode* B, TreeNode* t) {
        if(!A || A==t) return B;
        return getTargetCopy(A->left, B->left, t)?:getTargetCopy(A->right, B->right, t);
    }
};

File: 1379 - Find a Corresponding Node of a Binary Tree in a Clone of That Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def getTargetCopy(self, A: TreeNode, B: TreeNode, t: TreeNode) -> TreeNode:
        return B if not A or A == t else self.getTargetCopy(A.left, B.left, t) or self.getTargetCopy(A.right, B.right, t);